addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4763,47 +4762,76 @@ registry['sizing_x'] = registry.sizing.extend({
      * @override
      */
     _getSize: function () {
+        const isMobileView = weUtils.isMobile(this);
+        const resolutionModifier = isMobileView ? '' : 'lg-';
         var width = this.$target.closest('.row').width();
         var gridE = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
         var gridW = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11];
         this.grid = {
-            e: [_.map(gridE, v => ('col-lg-' + v)), _.map(gridE, v => width / 12 * v), 'width'],
-            w: [_.map(gridW, v => ('offset-lg-' + v)), _.map(gridW, v => width / 12 * v), 'margin-left'],
+            e: [
+                _.map(gridE, v => ('col-' + resolutionModifier + v)),
+                _.map(gridE, v => width / 12 * v),
+                'width'
+            ],
+            w: [
+                _.map(gridW, v => ('offset-' + resolutionModifier + v)),
+                _.map(gridW, v => width / 12 * v),
+                'margin-left'
+            ],
         };
         return this.grid;
     },
     /**
      * @override
      */
     _onResize: function (compass, beginClass, current) {
+        const isMobileView = weUtils.isMobile(this);
+        const resolutionModifier = isMobileView ? '' : 'lg-';
+        const resolutionModifierWithXL = isMobileView ? '' : '(?:lg-|xl-)';
         if (compass === 'w' || compass === 'e') {
-            const beginOffset = Number(beginClass.match(/offset-lg-([0-9-]+)|$/)[1] || beginClass.match(/offset-xl-([0-9-]+)|$/)[1] || 0);
+            const offsetRegexWithXL = new RegExp(`\\boffset-${resolutionModifierWithXL}([0-9-]+)|$`);
+            const offsetRegex = new RegExp(`\\boffset-${resolutionModifier}([0-9-]+)|$`);
+            const colRegex = new RegExp(`\\bcol-${resolutionModifier}(\\d{1,2})|$`);
+
+            const beginOffset = Number(beginClass.match(offsetRegexWithXL)[1] || 0);
+
+            const targetEl = this.$target[0];
```

Comment:
It's actually used 11 times in the if / else that follows, I think it's cleaner than repeating `this.$target[0]`. But I can remove it if you really think otherwise.

Reply:
Ok, I suppose it's a matter of taste